src/libostree/ostree-repo.c

```diff
@@ -3186,28 +3185,28 @@ reload_sysroot_config (OstreeRepo          *self,
                        GCancellable        *cancellable,
                        GError             **error)
 {
-  { g_autofree char *bootloader = NULL;
+  g_autofree char *bootloader = NULL;
 
-    if (!ot_keyfile_get_value_with_default_group_optional (self->config, "sysroot",
-                                                           "bootloader", "auto",
-                                                           &bootloader, error))
-      return FALSE;
-
-    /* TODO: possibly later add support for specifying a generic bootloader
-     * binary "x" in /usr/lib/ostree/bootloaders/x). See:
-     * https://github.com/ostreedev/ostree/issues/1719
-     * https://github.com/ostreedev/ostree/issues/1801
-     * Also, dedup these strings with the bootloader implementations
-     */
-    if (!(g_str_equal (bootloader, "auto") || g_str_equal (bootloader, "none")
-          || g_str_equal (bootloader, "zipl")))
-      return glnx_throw (error, "Invalid bootloader configuration: '%s'", bootloader);
+  if (!ot_keyfile_get_value_with_default_group_optional (self->config, "sysroot",
+                                                         "bootloader", "auto",
```

Comment:
I guess the `"auto"` here can be un-hardcoded in favor of the `CFG_SYSROOT_BOOTLOADER_OPTS_STR[]` entry?

Reply:
I've tried this and I think `CFG_SYSROOT_BOOTLOADER_OPTS_STR[CFG_SYSROOT_BOOTLOADER_OPT_AUTO]` is a rather awkward way to say "auto". It might make sense to add `ot_keyfile_get_enum_value_with_default` in the future and use it for "repo.mode" too, but I don't think it's worth it right now.

Reply:
Ack, understandable, I wasn't sure whether it was a simple leftover, I'm fine to keep it as is.